python/triton/compiler/compiler.py

```diff
@@ -83,6 +83,8 @@ def __init__(self, fn, signature, constants=None, attrs=None) -> None:
                     raise TypeError("Constants keys must be string")
         if self.attrs is None:
             self.attrs = AttrsDescriptor()
+        elif isinstance(self.attrs, dict):
+            self.attrs = AttrsDescriptor().from_dict(self.attrs)
```

Comment:
Should that be `AttrsDesriptor.from_dict(self.attrs)`?

Reply:
Yup, fixed.